lib/rouge/lexers/php.rb

```diff
@@ -69,104 +100,273 @@ def self.analyze_text(text)
         0
       end
 
+      ##### mixins to be included (DRY) #####
+
+      # recognize variables (in regular php and interpolated string)
+      state :common do
+        rule /\$#{LABEL}/, Name::Variable
+      end
+
+      DEFAULTS = Hash.new(Error).tap do |h|
+#         :in_scripting => Error,
+#         :var_offset => Error,
```

Comment:
Some states are "inherited" in few others, so to set the current token's type, I use this hash to map it to the correct type based on the current (top level) state's name. eg to highlight a `,` in `"...,..."` as `Str::Double` but in an expression like `"$a[,]"` as an error.

Do you have a better approach in mind?